### DISCUSSION TITLE
Sub-Module Releases

### DISCUSSION THREAD

**Discussion by alexanderbez** [goreleaser/goreleaser#2894] [Ideas, Issue Triage, and General discussions]
Goreleaser requires strict semantic versioning, which is typically a very good thing. However, sub-modules in Go are becoming more and more of a common approach to module management. E.g., You have a package/repo "foo" and a sub-module "bar", you'd tag and release the sub-module as bar/vX.Y.Z.

I noticed that in https://github.com/goreleaser/goreleaser/pull/2503, the `--skip-validate` removed the semantic versioning check bypass making it impossible to use goreleaser for sub-modules.

In our repo, this forced us to pin against v0.179.0 in order to have release workflows for sub-modules work, which isn't ideal (https://github.com/cosmos/cosmos-sdk/issues/11125).

Is there generally a better approach or even a plan to support sub-modules in the future?

**Comment by caarlos0** [comment]
it had to be done because other thing break down the way without it (e.g. linux packages).

it seems what you're doing is a monorepo, so you might want to check https://goreleaser.com/customization/monorepo/ out

**Reply by alexanderbez** [reply]
Woaa, this might be the answer I was looking for! Thank you so much.
